### PR TITLE
Adding More Furniture

### PR DESCRIPTION
**Pull request by WhaleyWhaley**
#### Summary
```SUMMARY: Content "Adds new furniture"```

#### Purpose of change
Adding 26 different pieces of **Furniture**, and 9 **items**, with a **recipe** and **snippet** type.

**Furniture Added** - 
**Charcoal Grill, Sandwich Board, Yard Flamingo, Birdfeeder, Traffic Cone, Taxidermy'd Bear, Foodperson Cardboard Cutout, Fishbowl, Dollhouse, Huge Teddy Bear, Drumset, TV Tray, Litterbox, Postal Boxes, Fire Hydrant, Floral Print Armchair, Beanbag Chair, Purple Velvet Chair, Plastic Lawn Chair, Foot Stool, Leather Chair, Gaming Chair, Wine Rack, Baby Gate, Turnstile, Door mats (With Snippets)**

The furniture should add some immersion or flavor. A lot of houses/buildings feel "empty" so hopefully this can be used in the future to add some depth to some locations. I also added the furniture to a few locations.  

Many of the furniture types are deployable, especially the decorative ones, because ideally more furniture would mean more base decoration and that way you can wield it and carry it away instead of dragging it.

#### Describe the solution
It just adds furniture.

#### Describe alternatives you've considered
The alternative is less furniture. 

#### Testing
Spawned each piece. The ones that can be deployed I deployed and undeployed. 

#### Additional context

Some sprites I made to save SDG some time.

fishbowlempty floralarmchair foodpersoncardboardalt_ sandwichboard_ trafficcone1_ tvtrayclosed tvtrayopen yardflamingo_ beartaxidermy2_ birdfeeder1_ birdfeeder2_ firehydrant fishbowl

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
You need to spawn your new furniture somewhere, otherwise it's a dead weight.

**Comment by WhaleyWhaley** [comment]
I mean, define "Dead Weight." They aren't going to harm anything and are fully available for mappers without making them define new furniture.

**Comment by Night-Pryanik** [comment]
We don't need stuff in the code (be it c++ or json) that isn't used anywhere in the game. In the past we had several cleanings of the code from unused stuff.
Make every type of your furniture spawn at least once somewhere, and it will be good to go.

File: data/json/furniture_and_terrain/furniture-appliances.json [hunk 1]
```diff
@@ -607,5 +607,26 @@
         { "item": "steel_chunk", "count": 3 }
       ]
     }
+  },
+  {
+    "type": "furniture",
+    "id": "f_charcoal_grill",
+    "name": "charcoal grill",
+    "symbol": "R",
+    "description": "A staple of the american backyard.  A simple metal container on legs built to hold charcoal for grilling.",
```
Comment (matskuman5) [inline data/json/furniture_and_terrain/furniture-appliances.json]:
```suggestion
    "description": "A staple of the American backyard.  A simple metal container on legs built to hold charcoal for grilling.",
```

File: data/json/furniture_and_terrain/furniture-decorative.json [hunk 2]
```diff
@@ -205,5 +205,299 @@
...
+      "sound_fail": "whump.",
+      "items": [ { "item": "2x4", "count": [ 1, 2 ] }, { "item": "nail", "charges": [ 2, 4 ] }, { "item": "splinter", "count": 2 } ]
+    }
+  },
+  {
+    "type": "furniture",
+    "id": "f_flamingo_decor",
+    "name": "flamingo decoration",
+    "symbol": "%",
+    "description": "Classy or Trashy?  You decide.",
```
Comment (matskuman5) [inline data/json/furniture_and_terrain/furniture-decorative.json]:
```suggestion
    "description": "Classy or trashy?  You decide.",
```

File: data/json/furniture_and_terrain/furniture-decorative.json [hunk 3]
```diff
@@ -205,5 +205,299 @@
...
+      "sound": "whump!",
+      "sound_fail": "thump.",
+      "items": [ { "item": "plastic_chunk", "count": [ 1, 2 ] } ]
+    }
+  },
+  {
+    "type": "furniture",
+    "id": "f_taxidermy_bear",
+    "name": "bear taxidermy statue",
+    "description": "A large Bear Taxidermy Statue.",
```
Comment (matskuman5) [inline data/json/furniture_and_terrain/furniture-decorative.json]:
```suggestion
    "description": "A large taxidermied bear.",
```

File: data/json/furniture_and_terrain/furniture-decorative.json [hunk 4]
```diff
@@ -205,5 +205,299 @@
...
+      "sound_fail": "clink.",
+      "items": [ { "item": "glass_shard", "charges": [ 2, 5 ] } ]
+    }
+  },
+  {
+    "type": "furniture",
+    "id": "f_dollhouse",
+    "name": "dollhouse",
+    "symbol": "A",
+    "description": "A well made scale home for dolls.",
```
Comment (matskuman5) [inline data/json/furniture_and_terrain/furniture-decorative.json]:
```suggestion
    "description": "A well-made scale home for dolls.",
```

File: data/json/furniture_and_terrain/furniture-decorative.json [hunk 5]
```diff
@@ -205,5 +205,299 @@
...
+      "sound": "thump!",
+      "sound_fail": "whack!",
+      "items": [ { "item": "rag", "count": [ 15, 25 ] }, { "item": "cotton_ball", "charges": [ 1, 5 ] } ]
+    }
+  },
+  {
+    "type": "furniture",
+    "id": "f_drumset",
+    "name": "drum set",
+    "description": "A full size drum set.",
```
Comment (matskuman5) [inline data/json/furniture_and_terrain/furniture-decorative.json]:
```suggestion
    "description": "A full-size drum set.",
```

File: data/json/furniture_and_terrain/furniture-seats.json [hunk 6]
```diff
@@ -204,5 +204,218 @@
       "sound_fail": "whump.",
       "items": [ { "item": "2x4", "count": [ 1, 3 ] }, { "item": "nail", "charges": [ 2, 6 ] }, { "item": "splinter", "count": 1 } ]
     }
+  },
+  {
+    "type": "furniture",
+    "id": "f_floralprint_armchair",
+    "name": "floral print arm chair",
```
Comment (matskuman5) [inline data/json/furniture_and_terrain/furniture-seats.json]:
```suggestion
    "name": "floral print armchair",
```

File: data/json/items/generic.json [hunk 7]
```diff
@@ -2896,5 +2896,139 @@
...
+  {
+    "id": "welcome_mat",
+    "type": "GENERIC",
+    "category": "other",
+    "symbol": "_",
+    "color": "brown",
+    "name": "welcome mat",
+    "name_plural": "welcome mats",
+    "snippet_category": "welcome_mats",
+    "description": "A mat from a homes front door.",
```
Comment (matskuman5) [inline data/json/items/generic.json]:
```suggestion
    "description": "A mat from a home's front door.",
```

File: data/json/snippets/snippets.json [hunk 8]
```diff
@@ -643,5 +643,20 @@
...
+    "type": "snippet",
+    "category": "welcome_mats",
+    "text": [
+      "'Welcome'",
+      "'Welcome Home'",
+      "'Beware Cats!'",
+      "'J O Y'",
+      "'U S A'",
+      "'An Animal Lover Lives Here'",
+      "It has an Image of flowers on it.",
```
Comment (matskuman5) [inline data/json/snippets/snippets.json]:
```suggestion
      "It has an image of flowers on it.",
```

File: data/json/snippets/snippets.json [hunk 9]
```diff
@@ -643,5 +643,20 @@
...
+      "It has an Image of skulls on it.",
```
Comment (matskuman5) [inline data/json/snippets/snippets.json]:
```suggestion
      "It has an image of skulls on it.",
```

File: data/json/snippets/snippets.json [hunk 10]
```diff
@@ -643,5 +643,20 @@
...
+      "It has an Image of a big heart on it and the word 'JOY'"
```
Comment (matskuman5) [inline data/json/snippets/snippets.json]:
```suggestion
      "It has an image of a big heart and the word 'JOY' on it"
```

File: data/json/furniture_and_terrain/furniture-decorative.json [hunk 11]
```diff
@@ -205,5 +205,299 @@
...
+        { "item": "wire", "charges": [ 1, 1 ] }
+      ]
+    }
+  },
+  {
+    "type": "furniture",
+    "id": "f_foodperson_cutout",
+    "name": "foodperson cutout",
+    "symbol": "8",
+    "description": "A large lifesize cutout of FoodPerson.",
```
Comment (DaviBones) [inline data/json/furniture_and_terrain/furniture-decorative.json]:
```suggestion
    "description": "A large life-size cutout of FoodPerson.",
```

File: data/json/furniture_and_terrain/furniture-decorative.json [hunk 12]
```diff
@@ -205,5 +205,299 @@
...
+    "color": "white",
+    "move_cost_mod": 2,
+    "coverage": 80,
+    "required_str": 1,
+    "flags": [ "FLAMMABLE_ASH", "ORGANIC", "SIGN" ],
+    "deconstruct": { "items": [ { "item": "paper", "count": 15 } ] },
```
Comment (DaviBones) [inline data/json/furniture_and_terrain/furniture-decorative.json]:
```suggestion
    "deconstruct": { "items": [ { "item": "paper", "charges": 15 } ] },
```

`"count": 15` would drop 750 paper, which is far too much.

File: data/json/furniture_and_terrain/furniture-decorative.json [hunk 13]
```diff
@@ -205,5 +205,299 @@
...
+    "bash": {
+      "str_min": 5,
+      "str_max": 40,
+      "sound": "smash!",
+      "sound_fail": "whump.",
+      "items": [ { "item": "paper", "count": [ 5, 15 ] } ]
```
Comment (DaviBones) [inline data/json/furniture_and_terrain/furniture-decorative.json]:
```suggestion
      "items": [ { "item": "paper", "charges": [ 5, 15 ] } ]
```

File: data/json/furniture_and_terrain/furniture-seats.json [hunk 14]
```diff
@@ -204,5 +204,218 @@
       "sound_fail": "whump.",
       "items": [ { "item": "2x4", "count": [ 1, 3 ] }, { "item": "nail", "charges": [ 2, 6 ] }, { "item": "splinter", "count": 1 } ]
     }
+  },
+  {
+    "type": "furniture",
+    "id": "f_floralprint_armchair",
+    "name": "floral print armchair",
+    "symbol": "H",
+    "description": "A more comfortable way of sitting down.  This reminds you of a nursing home.",
```
Comment (DaviBones) [inline data/json/furniture_and_terrain/furniture-seats.json]:
```suggestion
    "description": "A more comfortable way of sitting down.  It reminds you of a nursing home.",
```

Sounds more natural to me this way, feel free to leave it as is if you disagree.

File: data/json/furniture_and_terrain/furniture-seats.json [hunk 15]
```diff
@@ -204,5 +204,218 @@
...
+    "required_str": 7,
+    "deconstruct": { "items": [ { "item": "rag", "count": [ 25, 30 ] }, { "item": "cotton_ball", "charges": [ 2, 5 ] } ] },
+    "max_volume": 3000,
+    "flags": [ "TRANSPARENT", "FLAMMABLE_ASH", "ORGANIC", "PLACE_ITEM", "MOUNTABLE", "CAN_SIT", "SHORT" ],
+    "bash": {
+      "str_min": 8,
+      "str_max": 20,
+      "sound": "smash!",
+      "sound_fail": "whump.",
+      "items": [ { "item": "cotton_ball", "count": [ 2, 5 ] }, { "item": "rag", "count": [ 20, 30 ] } ]
```
Comment (DaviBones) [inline data/json/furniture_and_terrain/furniture-seats.json]:
```suggestion
      "items": [ { "item": "cotton_ball", "charges": [ 2, 5 ] }, { "item": "rag", "count": [ 20, 30 ] } ]
```

Assuming you meant it this way; as it is, it drops more cotton when smashed than when deconstructed.

**Comment by Outta-my-way-burrocrat** [comment]
Could you add theater seats to the movie theater in the mall?

mall_a_11 and mall_a_20

**Comment by Outta-my-way-burrocrat** [comment]
house_patio would be a good place for the charcoal grill.

**Comment by ZhilkinSerg** [comment]
Can you resolve conflicts please?

**Comment by Outta-my-way-burrocrat** [comment]
@WhaleyWhaley boop

**Comment by LilShiningMan** [comment]
Hoping this gets merged. Charcoal grill and drumset would be great to use!

**Comment by I-am-Erk** [comment]
I'd be happy to merge this if the conflicts were fixed, I can't do that easily on my phone though.

**Comment by Night-Pryanik** [comment]
@WhaleyWhaley care to resolve conflicts?

**Comment by ZhilkinSerg** [comment]
Need conflict resolution. Feel free to rebase and reopen.

**Pull request closed without merging**